### PR TITLE
Added Algolia search bar

### PR DESCRIPTION
**Pull request by lmammino**
As per title

kapture 2018-06-28 at 14 54 45

Not sure how/when the actual docuemnts will be indexed, I believe I can't see anything just yet because I am trying from localhost. @mcollina do you know more about this?

### CODE REVIEW

**Comment by mcollina** [comment]
You might be able to try this by adding a line to your `/etc/hosts` that resolves `www.fastify.io` to `localhost`. 

I'm also ok in landing this and redirect if this does not work.

**Comment by mcollina** [comment]
I'll also ping them to see!

**Comment by lmammino** [comment]
@mcollina, Let me try the localhost option and let's see what happens :)

**Comment by lmammino** [comment]
It doesn't work either. I have a feeling they wait to see they will scrape the original domain to see if their js file has been included as proof of ownership before starting to index all the content.

Do you want to give it a live try?

**Comment by mcollina** [comment]
let’s wait for them and see what they say!
Il giorno gio 28 giu 2018 alle 23:01 Luciano Mammino <
notifications@github.com> ha scritto:

**Comment by lmammino** [comment]
Cool ;)

Looking forward to see what they say!

**Comment by lmammino** [comment]
@mcollina, actually, it seems very likely we have to provide the indexing by ourselves. Ref docs [here](https://www.algolia.com/doc/tutorials/search-ui/autocomplete/auto-complete/).

If that's really the case, this is a bit of extra work that we have to figure out, but gives us the opportunity to be more in control of the indexing.

File: src/website/layouts/default.html [hunk 1]
```diff
@@ -18,6 +18,16 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
     <script>hljs.initHighlightingOnLoad();</script>
     {% block extraScripts %}{% endblock %}
+    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+    <script type="text/javascript">
+      docsearch({
+        apiKey: 'f7c62ddab40e653f67a6e15be761d951',
+        indexName: 'fastify',
+        inputSelector: '#algolia-search-input',
+        algoliaOptions: { 'facetFilters': ["version:$VERSION", "tags:$TAGS"] },
```
Comment (s-pace) [inline src/website/layouts/default.html]: You need to replace the `$VERSION` and `$TAGS`:

- Replace `$VERSION` with the version you want to search on.
   The list of possible version is hardcoded in the config.
   So as of today you have: `latest`, `master`

- Replace `$TAGS` with the tags you want to search on.
   The list of possible tags is hardcoded in the config.
   So as of today you have: `docs`, `ecosystem`, `home`

Comment (s-pace) [reply src/website/layouts/default.html]: For example if you want to refine the search to the version "latest" and the tags "docs" just specify:
```js
'facetFilters': ["version:latest", "tags:docs"]
```

**Comment by s-pace** [comment]
👋 

You might need to replace the $VERSION and $TAGS variable. Since the [`facetFilters` ](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/)are wrong, it reduces the scope to records matching "$VERSION" and "$TAGS", this is why you have no result.

I have tried it and everything should work as expected.

Let us know if you need anything.

**Review by s-pace — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
LGTM if it works

**Comment by lmammino** [comment]
Fantastic, thanks a million @s-pace, this works auto-magically™️ 🤩

kapture 2018-06-30 at 15 58 05

**Comment by lmammino** [comment]
@mcollina, check the GIF above, I'd be super happy to merge it! :)

**Comment by mcollina** [comment]
Go ahead! It’s fantastic!

**Comment by delvedor** [comment]
That's awesome, thanks!

**Pull request merged**